automated-testing/performance-testing/readme.md

```diff
@@ -1,67 +1,357 @@
+to have some buffer for both the kernel operations mentioned above as well as
+any other bursts of CPU required by other applications.
+
+### Queue Length
+
+This is the number of threads that are ready to execute but waiting for a core
+to become available.  On single core machines a sustained value greater than 2-3
+can mean that you have some CPU pressure.  Similarly, for a multicore machine
+divide the queue length by the number of cores and if that is continuously
+greater than 2-3 there might be CPU pressure.
+
+## Network Adapter
+
+Network speed is often a hidden culprit of poor performance. Finding the root
+cause to poor network performance is often difficult. The source of issues can
+originate from bandwidth hogs such as videoconferencing, transaction data,
+network backups, recreational videos.
+
+Packet loss is another area that contributes to poor performance. It occurs when
+packets fail to reach their destination, either by errors in the data
+transmission or network congestion. Packet loss can also be caused by Dos/DDoS
+attacks.
+
+Some of the tools that can help include:
+
+- ifconfig
+- netstat
+- iperf
+- tcpretrans
+- tcpdump
+- WireShark
+
+For most of these counters below, there are no hard and fast numbers. Wi-Fi
+speeds are lower as compared to wired, ethernet speeds. Networking speeds have
+increased dramatically over the years. We can expect all the numbers below to
+change over time.
+
+| Network Type | Speed  |
+|:-------------------- |:--------------------  |
+| Basic Wifi Speed | 3 to 8 Mpbs  |
+| Moderate Wifi Speed | 10 to 25 Mpbs  |
+| Fast Wifi Speed | 300 to 1000 Mbps  |
+| Fast Ethernet | 100 Mbps  |
+| Gigabit Ethernet | 1,000 Mbps  |
+| 10 Gigabit Ethernet | 10,000 Mbps  |
+
+### Network Counters
+
+The table above gives you some reference points to better understand what you
+can expect out of your network. Here are some counters that can help you
+understand where the bottlenecks might exist:
+
+### Bytes Received/sec
+
+The rate at which bytes are received over each network adapter.
+
+### Bytes Sent/sec
+
+The rate at which bytes are sent over each network adapter.
+
+### Bytes Total/sec
+
+The number of bytes sent and received over the network.
+
+### Segments Received/sec
+
+The rate at which segments are received for the protocol
+
+### Segments Sent/sec
+
+The rate at which segments are sent.
+
+### % Interrupt Time
+
+The percentage of time the processor spends receiving and servicing hardware
+interrupts. This value is an indirect indicator of the activity of devices that
+generate interrupts, such as network adapters.
+
+> There is an important distinction between **latency** and **throughput**.
+**Latency** measures the time it takes for a packet to be transferred across the
+network, either in terms of a one\-way transmission or a round\-trip
+transmission. **Throughput** is different and attempts to measure the quantity
+of data being sent and received within a unit of time.
```

Comment:
Since they all are one-liners, could you plz change them to bullet points or a table?

Reply:
Good idea. They now all look good as tables.